browser/js/cart/cart.factory.js

```diff
@@ -35,40 +48,85 @@ app.factory('CartFactory', function ($http, ProductFactory, localStorageService)
       cart.push(cartItem);
 
       syncLocalStorage();
-      return $http.post('/api/orders/' + orderId + '/item', cartItem)
-      .then( function(order) {
-        return order;
-      })
+      $rootScope.$broadcast('cart-updated');
+
+      // syncronize with database if logged in
+      if(isLoggedIn()) {
+        if(!orderId) { // if no pending order was found, create one
+          return createNewOrder()
+          .then(function() {
+            return $http.post('/api/orders/' + orderId + '/item', cartItem)
+          })
+          .then(res => res.data);
+        } else { // append to existing order
+          return $http.post('/api/orders/' + orderId + '/item', cartItem)
+          .then(res => res.data);
+        }
```

Comment:
we could do Order.findOrCreate({where: status: 'pending}) when the user is logged in and copy that to the cart to save a lot of duplicate code in the front-end  (such as this block, createNewOrder, and getPendingOrderDetails)

this is similar to the 'api/users/:userid/orders/pending' route currently